src/command/defaults/ClearCommand.php

```diff
@@ -110,65 +117,60 @@ public function execute(CommandSender $sender, string $commandLabel, array $args
+				$clearedCount += $this->countItems($inventory, null);
+				$inventory->clearAll();
 			}
-
-			$target->getInventory()->clearAll();
-			$target->getArmorInventory()->clearAll();
-			//TODO: should the cursor inv be cleared?
 		}else{
-			//clear the item from targets inventory irrelevant of the count
+			// Clear the item from target's inventory irrelevant of the count
 			if($maxCount === -1){
-				if(($slot = $target->getArmorInventory()->first($item)) !== -1){
-					$cleared++;
-					$target->getArmorInventory()->clear($slot);
-				}
-
-				foreach($target->getInventory()->all($item) as $index => $i){
-					$cleared += $i->getCount();
-					$target->getInventory()->clear($index);
+				foreach($inventories as $inventory) {
+					$clearedCount += $this->countItems($inventory, $targetItem);
+					$inventory->remove($targetItem);
 				}
 			}else{
-				//clear only the given amount of that particular item from targets inventory
-				if(($slot = $target->getArmorInventory()->first($item)) !== -1){
-					$cleared++;
-					$maxCount--;
-					$target->getArmorInventory()->clear($slot);
-				}
-
-				if($maxCount > 0){
-					foreach($target->getInventory()->all($item) as $index => $i){
-						if($i->getCount() >= $maxCount){
-							$i->pop($maxCount);
-							$cleared += $maxCount;
-							$target->getInventory()->setItem($index, $i);
-							break;
-						}
+				// Clear the item from target's inventory up to the count
+				$inventoryIndex = 0;
+				while($maxCount > 0 && $inventoryIndex < count($inventories)){
+					$inventory = $inventories[$inventoryIndex];
+					// Move onto next inventory from prioritization list if empty
+					if(count($inventory->getContents()) === 0){
+						$inventoryIndex++;
+						continue;
+					}
+					foreach($inventory->all($targetItem) as $index => $item) {
+						$count = min($item->getCount(), $maxCount);
+						$item->pop($count);
+						$clearedCount += $count;
+						$inventory->setItem($index, $item);
 
+						$maxCount -= $count;
 						if($maxCount <= 0){
 							break;
 						}
-
-						$cleared += $i->getCount();
-						$maxCount -= $i->getCount();
-						$target->getInventory()->clear($index);
 					}
+					$inventoryIndex++;
 				}
 			}
 		}
 
-		if($cleared > 0){
-			Command::broadcastCommandMessage($sender, KnownTranslationFactory::commands_clear_success($target->getName(), (string) $cleared));
+		if($clearedCount > 0){
+			Command::broadcastCommandMessage($sender, KnownTranslationFactory::commands_clear_success($target->getName(), (string) $clearedCount));
 		}else{
 			$sender->sendMessage(KnownTranslationFactory::commands_clear_failure_no_items($target->getName())->prefix(TextFormat::RED));
 		}
 
 		return true;
 	}
-}
+
+	protected function countItems(SimpleInventory $inventory, ?Item $target): int {
+		return array_reduce(
+			$target instanceof Item ? $inventory->all($target) : $inventory->getContents(),
+			static fn(int $carry, Item $item): int => $carry + $item->getCount(),
```

Comment:
```suggestion
			static fn(int $carry, Item $item) => $carry + $item->getCount(),
```

The return type is unnecessary boilerplate (it can be inferred from the return expression and the return expr can't be forgotten).